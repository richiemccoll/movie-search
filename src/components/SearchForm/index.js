import React from "react";

function SearchForm({ handleSearchSubmit, error }) {
  const inputRef = React.createRef();
  return (
    <section>
      <form
        onSubmit={(e) => {
          e.preventDefault();
          handleSearchSubmit(inputRef.current.value);
        }}
      >
        <div role="search">
          <label className="search-label" htmlFor="search">
            Search for a movie
            {error ? (
              <p className="error" role="alert">
                {error}
              </p>
            ) : null}
          </label>

          <input
            id="search"
            className="search-input"
            type="text"
            ref={inputRef}
            aria-invalid={!!error}
            required
          />
          <input className="button" type="submit" value="Search" />
        </div>
        <style jsx>
          {`
            form {
              border: 1px solid rgba(0, 0, 0, 0.15);
              border-radius: 5px;
              padding: 50px;
            }
            .search-label {
              display: block;
              font-size: 18px;
              margin-bottom: 5px;
            }

            .search-input {
              display: block;
              width: 100%;
              font-size: 16px;
              border-radius: 5px;
              border: 2px solid grey;
              -webkit-appearance: none;
              padding: 1em 0.5em;
              margin-bottom: 8px;
              box-sizing: border-box;
            }

            .error {
              display: block;
              background-color: #f44336;
              padding: 10px;
              border-radius: 5px;
            }

            .button {
              background: #2979ff;
              color: #fff;
              border: 0;
              border-radius: 5px;
              appearance: none;
              cursor: pointer;
              font-size: 16px;
              padding: 1em 2.5em;
              border: 2px solid transparent;
            }
          `}
        </style>
      </form>
    </section>
  );
}

export default SearchForm;
