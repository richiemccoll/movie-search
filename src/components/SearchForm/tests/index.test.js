import React from "react";
import renderer from "react-test-renderer";

import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SearchForm from "..";

Enzyme.configure({ adapter: new Adapter() });

describe("SearchForm", () => {
  it("should render correctly", () => {
    const component = renderer.create(<SearchForm />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call handleSubmit with the value of the input field", () => {
    const searchSubmitSpy = jest.fn();
    const initial = mount(<SearchForm handleSearchSubmit={searchSubmitSpy} />);
    initial.find("#search").simulate("change", { target: { value: "test " } });
    initial.find("form").simulate("submit", { preventDefault: Function.prototype });
    expect(searchSubmitSpy).toBeCalled();
  });

  it("should display some sort of error to the user", () => {
    const initial = shallow(<SearchForm />);
    expect(initial.find(".error").length).toBe(0);
    expect(initial.find("#search").html()).toMatchSnapshot();

    const result = initial.setProps({ error: "This is a message " });
    expect(result.find(".error").length).toBe(1);
    expect(initial.find("#search").html()).toMatchSnapshot();
  });
});
