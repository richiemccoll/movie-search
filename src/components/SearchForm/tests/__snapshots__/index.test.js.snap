// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should display some sort of error to the user 1`] = `"<input type=\\"text\\" id=\\"search\\" aria-invalid=\\"false\\" required=\\"\\" class=\\"jsx-4134789409 search-input\\"/>"`;

exports[`SearchForm should display some sort of error to the user 2`] = `"<input type=\\"text\\" id=\\"search\\" aria-invalid=\\"true\\" required=\\"\\" class=\\"jsx-4134789409 search-input\\"/>"`;

exports[`SearchForm should render correctly 1`] = `
<section>
  <form
    className="jsx-4134789409"
    onSubmit={[Function]}
  >
    <div
      className="jsx-4134789409"
      role="search"
    >
      <label
        className="jsx-4134789409 search-label"
        htmlFor="search"
      >
        Search for a movie
      </label>
      <input
        aria-invalid={false}
        className="jsx-4134789409 search-input"
        id="search"
        required={true}
        type="text"
      />
      <input
        className="jsx-4134789409 button"
        type="submit"
        value="Search"
      />
    </div>
  </form>
</section>
`;
